src/agent/sqlite_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,289 @@
+                std::vector<Col> fields;
+                fields.reserve(nColumns);
+                for (int i = 0; i < nColumns; i++)
+                {
+                    sqlite_manager::Col field(
+                        query.getColumn(i).getName(), query.getColumn(i).getType(), query.getColumn(i).getString());
+                    fields.push_back(field);
+                }
+                results.push_back(fields);
+            }
+
+            return results;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error during Retrieve operation: " << e.what() << std::endl;
+            return {};
+        }
+        return results;
+    }
+
+    void SQLiteManager::Remove(const std::string& tableName, const std::vector<Col>& selCriteria)
+    {
+        // Build the query string
+        std::string queryString = std::format("DELETE FROM {}", tableName);
+        if (selCriteria.size() != 0)
+        {
+            queryString += " WHERE ";
+            for (auto& col : selCriteria)
+            {
+                if (col.m_type == SQLite::TEXT)
+                {
+                    queryString += col.m_name + "='" + col.m_value + "' AND ";
+                }
+                else
+                {
+                    queryString += col.m_name + "=" + col.m_value + " AND ";
+                }
+            }
+            // Remove the trailing " AND "
+            queryString.erase(queryString.size() - 5);
+        }
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error executing SQL: " << e.what() << std::endl;
+            throw;
+        }
+    }
+
+    void SQLiteManager::Update(const std::string& tableName,
+                               const std::vector<Col>& fields,
+                               const std::vector<Col>& selCriteria)
+    {
+        if (fields.size() == 0)
+        {
+            std::cerr << "Error: Missing update fields" << std::endl;
+            throw;
+        }
+
+        // Build the query string
+        std::string queryString = std::format("UPDATE {} SET ", tableName);
+
+        for (auto& col : fields)
+        {
+            if (col.m_type == SQLite::TEXT)
+            {
+                queryString += col.m_name + "='" + col.m_value + "', ";
+            }
+            else
+            {
+                queryString += col.m_name + "=" + col.m_value + ", ";
+            }
+        }
+
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 2);
+
+        if (selCriteria.size() != 0)
+        {
+            queryString += " WHERE ";
+            for (auto& col : selCriteria)
+            {
+                if (col.m_type == SQLite::TEXT)
+                {
+                    queryString += col.m_name + "='" + col.m_value + "' AND ";
```

Comment:
Same question.

Reply:
Added extra parameter that allows the caller to specify the operator AND or OR. Default is AND.